qubes/events.py

```diff
@@ -115,6 +115,17 @@ def add_handler(cls, event, func):
         # pylint: disable=no-member
         cls.__handlers__[event].add(func)
 
+    @classmethod
+    def remove_handler(cls, event, func):
+        '''Remove event handler from subject's class.
+
+        :param str event: event identificator
+        :param collections.Callable handler: handler callable
+        '''
+
+        # pylint: disable=no-member
+        cls.__handlers__[event].remove(func)
+
```

Comment:
We can have filters in Dispatcher:

```python
def register(self, connection, subject, filters):
  self._connections[subject].append((connection, filters))
def _handler(self, subject, event, **kwargs):
  for connection, filters in self._connections[subject]:
    if not tuple(apply_filters([(subject, event, kwargs)], filters)):
      continue
    connection.send_event(subject, event, **kwargs)

No, it's not possible to detect that. If you call ``obj.clsmethod`` (when clsmethod is decorated with ``@classmethod``), you're really calling ``type(obj).clsmethod``. That is, unless you'd like to reimplement ``classmethod``. It would be possible to have per-instance events instead, by having __handlers__ per-instance, and initialised in ``__new__`` with anything the class wishes.